docs/csharp/language-reference/operators/patterns.md

```diff
@@ -145,7 +145,9 @@ As the preceding example shows, you can repeatedly use the pattern combinators i
 
 ### Precedence and order of checking
 
-The following list orders pattern combinators starting from the highest precedence to the lowest:
+A pattern that is part of an `is` expression has *relational* precedence. It's the same precedence as the `is` operator. Pattern expressions in the `case` arms of a `switch` statement have the same precedence as the *conditional logical* operators (`||` and `&&`). Pattern expressions inside a `switch` expression have the same precedence as the coalescing operators (`??` and `throw` expressions).
+
+The individual pattern expressions are evaluated in order, except as noted for the pattern combinators. The pattern combinators are ordered from the highest precedence to the lowest as shown in the following list:
```

Comment:
The new version looks to contradict the note below (again this note talks about logical patterns only):

> [!NOTE]
> The order in which patterns are checked is undefined. At run time, the right-hand nested patterns of `or` and `and` patterns can be checked first.

I'm not sure the added sentence helps. What is a "pattern expression"? In which order? If there are no pattern combinators, then things look much simpler and I don't think they worth mentioning.

Again this section talks about only logical patterns, so the introduction of other patterns is out-of-context.

Reply:
Given https://github.com/dotnet/docs/pull/36065#discussion_r1249759802 I suggest to revert this change and add the new proposed paragraph after the precedence list of `not`/`and/`or`.